### PR TITLE
[ML] Log best hyperparameters and runtime summary

### PR DESCRIPTION
**Pull request by valeriy42**
This PR changes the logging level for the best hyperparameters from TRACE to INFO. It also adds tracking runtime for the hyperparameter optimization steps and logs the total number of steps, mean runtime per iteration and its stdev.

### CODE REVIEW

File: lib/maths/CBoostedTreeImpl.cc [hunk 1]
```diff
@@ -485,6 +493,12 @@ void CBoostedTreeImpl::train(core::CDataFrame& frame,
         m_BestForest = this->trainForest(frame, this->allTrainingRowsMask(), recordMemoryUsage);
         this->recordState(recordTrainStateCallback);
 
+        timeAccumulator.add(stopWatch.stop());
+
+        LOG_INFO(<< "Training finished after " << m_CurrentRound << " iterations. Time per iteration in ms mean: "
+                 << timeAccumulator.s_Moments[0]
```
Comment (tveasey) [inline lib/maths/CBoostedTreeImpl.cc]:
```suggestion
                 << CBasicStatistics::mean(timeAccumulator)
```

File: lib/maths/CBoostedTreeImpl.cc [hunk 2]
```diff
@@ -485,6 +493,12 @@ void CBoostedTreeImpl::train(core::CDataFrame& frame,
...
+                 << " std. dev:  " << std::sqrt(timeAccumulator.s_Moments[1]));
```
Comment (tveasey) [inline lib/maths/CBoostedTreeImpl.cc]: This is the unbiased variance estimate and what we'd normally use
```suggestion
                 << " std. dev:  " << std::sqrt(CBasicStatistics::variance(timeAccumulator)));
```

File: lib/maths/CBoostedTreeImpl.cc [hunk 3]
```diff
@@ -444,6 +445,11 @@ void CBoostedTreeImpl::train(core::CDataFrame& frame,
         LOG_TRACE(<< "Test loss = " << m_BestForestTestLoss);
 
     } else if (m_CurrentRound < m_NumberRounds || m_BestForest.empty()) {
+        using TUIntMeanVarAccumulator =
+            CBasicStatistics::SSampleMeanVar<std::uint64_t>::TAccumulator;
```
Comment (tveasey) [inline lib/maths/CBoostedTreeImpl.cc]: This doesn't like integers. We could do something to manage integer division better, but it is probably better to stick to double and cast when adding the iteration time.

File: lib/maths/CBoostedTreeImpl.cc (hunk 1 above)
Comment (valeriy42) [reply lib/maths/CBoostedTreeImpl.cc]: Oh nice! I learned something new today :D

File: lib/maths/CBoostedTreeImpl.cc [hunk 4]
```diff
@@ -444,6 +445,11 @@ void CBoostedTreeImpl::train(core::CDataFrame& frame,
         LOG_TRACE(<< "Test loss = " << m_BestForestTestLoss);
 
     } else if (m_CurrentRound < m_NumberRounds || m_BestForest.empty()) {
+        using TUIntMeanVarAccumulator =
```
Comment (tveasey) [inline lib/maths/CBoostedTreeImpl.cc]: nit: this should be renamed to something like TFloatMeanVarAccumulator. (I'd also be inclined to just go with double: there is no real reason to save memory or runtime here.)

**Pull request merged**